dace/sdfg/validation.py

```diff
@@ -191,6 +191,22 @@ def validate_state(state: 'dace.sdfg.SDFGState',
         except Exception as ex:
             raise InvalidSDFGNodeError("Node validation failed: " + str(ex), sdfg, state_id, nid) from ex
 
+        if Config.get_bool('check_race_conditions'):
+            if isinstance(node, nd.AccessNode):
+                incoming_edge_memlet_ranges = []
+                for e in state.in_edges(node):
+                    memlet_range = str(e.data.subset).split(':')
+                    incoming_edge_memlet_ranges.append([int(memlet_range[0]), int(memlet_range[1])])
+                if (len(incoming_edge_memlet_ranges) > 0):
+                    sorted(incoming_edge_memlet_ranges)
+                    prev_max = incoming_edge_memlet_ranges[0][1]
+                    for p in incoming_edge_memlet_ranges[1:]:
+                        if (p[0] - prev_max < 0):
+                            warnings.warn('WARNING: Memlet range overlap while writing in "%s" in state %s' %
```

Comment:
```suggestion
                            warnings.warn('Memlet range overlap while writing in "%s" in state %s' %
```
The word "WARNING" is redundant :)